src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClientExceptions.cs

```diff
@@ -7,7 +7,7 @@ namespace Microsoft.Diagnostics.NETCore.Client
 {
     public class DiagnosticsClientException : Exception
     {
-        public DiagnosticsClientException(string msg) : base(msg) {}
+        public DiagnosticsClientException(string msg, Exception exception=null) : base(msg, exception) {}
```

Comment:
This is a binary breaking change on a public type. I would just create a separate exception.

Reply:
I'd propose a new overload on the constructor, not a new Exception:

```suggestion
        public DiagnosticsClientException(string msg) : base(msg) {}
        public DiagnosticsClientException(string msg, Exception exception) : base(msg, exception) {}
```

Reply:
I was meaning to say that as I read the argument name 😆 thanks for fact checking me here.